export_kakuma_advert_contacts.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import csv
+import json
+from collections import OrderedDict
+
+from core_data_modules.logging import Logger
+from core_data_modules.cleaners import Codes
+from core_data_modules.traced_data.io import TracedDataJsonIO
+from id_infrastructure.firestore_uuid_table import FirestoreUuidTable
+from storage.google_cloud import google_cloud_utils
+
+from src.lib import PipelineConfiguration
+from src.lib.code_schemes import CodeSchemes
+
+Logger.set_project_name("WUSC-KEEP-II")
+log = Logger(__name__)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Generates a list of phone numbers and textit language of "
+                                                 "WUSC-KEEP-II-KAKUMA participants"
+                                                 "based on their manually labeled demographic language response")
+
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to WUSC-KEEP-II-KAKUMA pipeline configuration json file")
+    parser.add_argument("messages_json_input_path", metavar="messages_json_input_path",
+                        help="Path to the WUSC-KEEP-II-KAKUMA messages traced data JSONL file to extract phone"
+                             "numbers from")
+    parser.add_argument("contacts_csv_path", metavar="contacts-csv-path",
+                        help="CSV file path to write per-language contacts data to")
+
+    args = parser.parse_args()
+
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    messages_json_input_path = args.messages_json_input_path
+    contacts_csv_path = args.contacts_csv_path
+
+    # Read the settings from the configuration file
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+        assert pipeline_configuration.pipeline_name == "kakuma_pipeline", "PipelineName must be kakuma_pipeline"
+
+    log.info("Downloading Firestore UUID Table credentials...")
+    firestore_uuid_table_credentials = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path,
+        pipeline_configuration.phone_number_uuid_table.firebase_credentials_file_url
+    ))
+
+    phone_number_uuid_table = FirestoreUuidTable(
+        pipeline_configuration.phone_number_uuid_table.table_name,
+        firestore_uuid_table_credentials,
+        "avf-phone-uuid-"
+    )
+    log.info("Initialised the Firestore UUID table")
+
+    # Read the individuals dataset
```

Comment:
```suggestion
    # Read the messages dataset
```

In theory you could accept either dataset, provided there are labels in the individuals dataset. As this will likely happen in the near-future, you could future-proof by saying "Read the dataset", "Loaded n objects from the dataset" etc.

Reply:
updated thanks!